src/Raven.Server/Documents/Handlers/RevisionsHandler.cs

```diff
@@ -209,6 +210,97 @@ public async Task GetRevisionsFor()
             }
         }
 
+        [RavenAction("/databases/*/revisions/size", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
+        public async Task GetRevisionsSize()
+        {
+            List<RevisionSizeDetails> sizes;
+            List<string> changeVectors;
+
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), "ChangeVectors");
+                var parameters = JsonDeserializationServer.Parameters.GetRevisionsSizeParameters(json);
+
+                using (context.OpenReadTransaction())
+                using (var token = CreateHttpRequestBoundOperationToken())
+                {
+                    changeVectors = parameters.ChangeVectors;
+                    sizes = GetRevisionsSizeByChangeVector(context, changeVectors);
+                }
+            }
+
+            var actualEtag = ComputeHttpEtags.ComputeEtagForRevisionsSize(changeVectors, sizes);
+
+            var etag = GetStringFromHeaders(Constants.Headers.IfNoneMatch);
+            if (etag == actualEtag)
+            {
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+                return;
+            }
+
+            HttpContext.Response.Headers[Constants.Headers.Etag] = "\"" + actualEtag + "\"";
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext ctx))
+            await using (var writer = new AsyncBlittableJsonTextWriter(ctx, ResponseBodyStream()))
+            {
+                writer.WriteStartObject();
+
+                writer.WritePropertyName("Sizes");
+
+                writer.WriteStartArray();
```

Comment:
there is a WriteArray extension method, can we use it?

Reply:
I added an WriteArray overload, because there is no array for `IEnumerable<DynamicJsonValues>`